app/src/main/java/com/example/perfectweatherallyear/util/ForecastNotifierWorker.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.perfectweatherallyear.util
+
+import android.content.Context
+import androidx.work.*
+import androidx.work.ListenableWorker.Result.failure
+import androidx.work.ListenableWorker.Result.success
+import com.example.perfectweatherallyear.appComponent
+import com.example.perfectweatherallyear.model.DayWeather
+import com.example.perfectweatherallyear.model.Location
+import com.example.perfectweatherallyear.repository.DataResult
+import com.example.perfectweatherallyear.repository.WeatherRepository
+import com.example.perfectweatherallyear.ui.weekWeather.DAYS_NUMBER
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class ForecastNotifierWorker(appContext: Context, workerParams: WorkerParameters) :
+    CoroutineWorker(appContext, workerParams) {
+    @Inject
+    lateinit var repository: WeatherRepository
+
+    init {
+        appContext.appComponent.inject(this)
+    }
+
+    override suspend fun doWork(): Result {
+
+        return when (val hourlyWeatherForNotificationResult = loadHourWeatherTempForNotification()) {
+            is DataResult.Ok -> {
+                NotificationHandler.postNotification(applicationContext, hourlyWeatherForNotificationResult.response)
+                success()
+            }
+            is DataResult.Error -> {
+                hourlyWeatherForNotificationResult.error
+                failure()
+            }
+        }
+    }
+
+    private suspend fun loadDayWeatherForNotification(): DayWeather? {
```

Comment:
Remove from naming ForNotification. Abstract from details of the further data handling